libfwupd/fwupd-remote.c

```diff
@@ -110,15 +117,21 @@ fwupd_remote_set_url (FwupdRemote *self, const gchar *url)
 	if (self->uri == NULL)
 		return;
 
-	/* generate the signature URI too */
-	url_asc = g_strdup_printf ("%s.asc", url);
-	self->uri_asc = fwupd_remote_build_uri (self, url_asc, NULL);
-
 	/* generate some plausible local filenames */
 	basename = g_path_get_basename (soup_uri_get_path (self->uri));
 	self->filename = g_strdup_printf ("%s-%s", self->id, basename);
-	basename_asc = g_path_get_basename (soup_uri_get_path (self->uri_asc));
-	self->filename_asc = g_strdup_printf ("%s-%s", self->id, basename_asc);
+
+	/* generate the signature URI too */
+	if (self->keyring_kind == FWUPD_KEYRING_KIND_GPG) {
+		url_asc = g_strdup_printf ("%s.asc", url);
+	} else if (self->keyring_kind == FWUPD_KEYRING_KIND_PKCS7) {
+		url_asc = g_strdup_printf ("%s.p7b", url);
+	}
+	if (url_asc != NULL) {
+		self->uri_asc = fwupd_remote_build_uri (self, url_asc, NULL);
```

Comment:
How about introduce `fwupd_remote_get_uri_sig `and make `fwupd_remote_get_uri_asc `just call that.  Deprecate `fwupd_remote_get_uri_asc `to be dropped at some point.